uint8 hw_patch
uint8 hw_minor
uint8 hw_major
